internal/apmlog/logger.go

```diff
@@ -52,17 +52,24 @@ var (
 	}
 )
 
-func init() {
-	InitDefaultLogger()
-}
+// DefaultLogger initialises defaultLogger using the environment variables
+// ELASTIC_APM_LOG_FILE and ELASTIC_APM_LOG_LEVEL. If defaultLogger is non-nil,
+// it returns the logger.
+func DefaultLogger() *LevelLogger {
+	loggerMu.RLock()
+	if defaultLogger != nil {
+		defer loggerMu.RUnlock()
+		return defaultLogger
+	}
+	loggerMu.RUnlock()
+
+	loggerMu.Lock()
+	defer loggerMu.Unlock()
```

Comment:
Should we check if defaultLogger is non-nil again after taking the exclusive lock?
Otherwise, it could happen that another goroutine calls SetDefaultLogger just after the read lock is released here, but before the exclusive lock is taken.

Reply:
yeah that's a good point